sklearn/neighbors/tests/test_sample_weight.py

```diff
@@ -0,0 +1,55 @@
+import numpy as np
+import pytest
+import scipy.stats
+
+from sklearn import neighbors
+from sklearn.utils._testing import assert_array_almost_equal
+
+
+@pytest.mark.parametrize("weights", ["uniform", "distance", scipy.stats.norm.pdf])
+def test_KNeighborsRegressor_sample_weight_none_is_same_as_uniform(
+    weights,
+    n_samples=40,
+    n_features=5,
+    n_test_pts=10,
+    n_neighbors=3,
+    random_state=0,
+):
+    rng = np.random.RandomState(random_state)
+    X = 2 * rng.rand(n_samples, n_features) - 1
+    y = np.sqrt((X ** 2).sum(1))
+    y /= y.max()
+
+    epsilon = 1e-5 * (2 * rng.rand(1, n_features) - 1)
+
+    def _predict(sample_weight):
+        knn = neighbors.KNeighborsRegressor(n_neighbors=n_neighbors, weights=weights)
+        knn.fit(X, y, sample_weight)
+        return knn.predict(X[:n_test_pts] + epsilon)
+
+    y_pred_sample_weight_none = _predict(None)
```

Comment:
I would avoid creating a function here. It should be possible to write the same in few line and maybe parametrize it?

Reply:
ok